logdna/resource_archive.go

```diff
@@ -0,0 +1,278 @@
+package logdna
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const archiveConfigID = "archive"
+
+type archiveConfig struct {
+	Integration        string `json:"integration"`
+	Bucket             string `json:"bucket,omitempty"`
+	Endpoint           string `json:"endpoint,omitempty"`
+	Apikey             string `json:"apikey,omitempty"`
+	Resourceinstanceid string `json:"resourceinstanceid,omitempty"`
+	Accountname        string `json:"accountname,omitempty"`
+	Accountkey         string `json:"accountkey,omitempty"`
+	Projectid          string `json:"projectid,omitempty"`
+	Space              string `json:"space,omitempty"`
+	Accesskey          string `json:"accesskey,omitempty"`
+	Secretkey          string `json:"secretkey,omitempty"`
+	Authurl            string `json:"authurl,omitempty"`
+	Expires            int    `json:"expires,omitempty"`
+	Username           string `json:"username,omitempty"`
+	Password           string `json:"password,omitempty"`
+	Tenantname         string `json:"tenantname,omitempty"`
+}
+
+func resourceArchiveConfigCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	pc := m.(*providerConfig)
+	c := archiveConfig{
+		Integration:        d.Get("integration").(string),
+		Bucket:             d.Get("bucket").(string),
+		Endpoint:           d.Get("endpoint").(string),
+		Apikey:             d.Get("apikey").(string),
+		Resourceinstanceid: d.Get("resourceinstanceid").(string),
+		Accountname:        d.Get("accountname").(string),
+		Accountkey:         d.Get("accountkey").(string),
+		Projectid:          d.Get("projectid").(string),
+		Space:              d.Get("space").(string),
+		Accesskey:          d.Get("accesskey").(string),
+		Secretkey:          d.Get("secretkey").(string),
+		Authurl:            d.Get("authurl").(string),
+		Expires:            d.Get("expires").(int),
+		Username:           d.Get("username").(string),
+		Password:           d.Get("password").(string),
+		Tenantname:         d.Get("tenantname").(string),
+	}
+
+	req := newRequestConfig(
+		pc,
+		"POST",
+		"/v1/config/archiving",
+		c,
+	)
+
+	body, err := req.MakeRequest()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cn := archiveConfig{}
+	err = json.Unmarshal(body, &cn)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(archiveConfigID)
+	appendError(d.Set("integration", cn.Integration), &diags)
```

Comment:
Nit: similar to the comment above, this code block can be modularized to only set the appropriate JSON fields based on the `Integration`

Reply:
changed!